airflow/api_connexion/parameters.py

```diff
@@ -20,11 +20,11 @@
 from pendulum.parsing import ParserError
 
 from airflow.api_connexion.exceptions import BadRequest
+from airflow.configuration import conf
 from airflow.utils import timezone
 
-# Page parameters
-page_offset = "offset"
-page_limit = "limit"
+MAXIMUM_PAGE_LIMIT = 1000
```

Comment:
Why is there a hard limit in the code?  I think we can trust the user that he will not configure the limit incorrectly. I am able to imagine situations that this limit may be insufficient if the user wants to migrate data from one instance to another.